internal/v1/server_test.go

```diff
@@ -321,6 +321,88 @@ func TestGetComposeStatus404(t *testing.T) {
 	require.Contains(t, body, "404 during tests")
 }
 
+func TestGetComposeMetadata(t *testing.T) {
+	// simulate osbuild-composer API
+	testPackages := []PackageMetadata{
+		{
+			Arch:      "ArchTest2",
+			Epoch:     strptr("EpochTest2"),
+			Name:      "NameTest2",
+			Release:   "ReleaseTest2",
+			Sigmd5:    "Sigmd5Test2",
+			Signature: strptr("SignatureTest2"),
+			Type:      "TypeTest2",
+			Version:   "VersionTest2",
+		},
+		{
+			Arch:      "ArchTest1",
+			Epoch:     strptr("EpochTest1"),
+			Name:      "NameTest1",
+			Release:   "ReleaseTest1",
+			Sigmd5:    "Sigmd5Test1",
+			Signature: strptr("SignatureTest1"),
+			Type:      "TypeTest1",
+			Version:   "VersionTest1",
+		},
+	}
+	api_srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		m := ComposeMetadata{
+			OstreeCommit: strptr("test string"),
+			Packages:     &testPackages,
+		}
+
+		err := json.NewEncoder(w).Encode(m)
+		require.NoError(t, err)
+	}))
+
+	defer api_srv.Close()
+
+	srv := startServer(t, api_srv.URL, "*", "")
+
+	defer func() {
+		err := srv.Shutdown(context.Background())
+		require.NoError(t, err)
+	}()
+
+	var result ComposeMetadata
+
+	// Get API response and compare
+	response, body := tutils.GetResponseBody(t, "http://localhost:8086/api/image-builder/v1/composes/xyz-123-test/metadata", &tutils.AuthString0)
+	require.Equal(t, 200, response.StatusCode)
+	err := json.Unmarshal([]byte(body), &result)
+	require.NoError(t, err)
+	require.Equal(t, *result.Packages, testPackages)
+
+	// With a wildcard orgIds either auth should work
+	response, body = tutils.GetResponseBody(t, "http://localhost:8086/api/image-builder/v1/composes/xyz-123-test/metadata", &tutils.AuthString1)
+	require.Equal(t, 200, response.StatusCode)
+	err = json.Unmarshal([]byte(body), &result)
+	require.NoError(t, err)
+	require.Equal(t, *result.Packages, testPackages)
```

Comment:
We already test the identityfilter in several places, it's a bit redundant to test with both AuthString0 and AuthString1 here.